### PR TITLE
Upgrade recipe for parmetis to be a CMakePackage

### PR DESCRIPTION
**Pull request by KineticTheory**
+ Eliminate `install` method (use the one from CMakePackage).
+ Move configure options to new method `cmake_args` and eliminate the `+debug` variant as it might conflict with `build_type=` and is redundant with `build_type=Debug +gdb`.
+ Move special install instructions for DarwinOS to a `run_after` method.

This needs to be tested on Darwin OS before being merged.  I am unable to test the `darwin_fix` method.

### CODE REVIEW

File: var/spack/repos/builtin/packages/parmetis/package.py [hunk 1]
```diff
@@ -93,11 +92,10 @@ def install(self, spec, prefix):
...
-            make('install')
+        return options
 
-            # The shared library is not installed correctly on Darwin; fix this
-            if (sys.platform == 'darwin') and ('+shared' in spec):
-                fix_darwin_install_name(prefix.lib)
+    @run_after('install')
+    def darwin_fix(self):
+        # The shared library is not installed correctly on Darwin; fix this
+        if (sys.platform == 'darwin') and ('+shared' in spec):
```
Comment (davydden) [inline var/spack/repos/builtin/packages/parmetis/package.py]: spec is not defined here

Comment (KineticTheory) [reply var/spack/repos/builtin/packages/parmetis/package.py]: Thanks for catching this defect.  Some of the Travis checks failed as well.  I'll work on fixing those and push an update.

**Comment by KineticTheory** [comment]
@davydden Similar to my changes in `metis/package.py` (#6808), should I remove the `+debug` variant?

**Comment by adamjstewart** [comment]
Closing and reopening to restart the Travis tests.

**Comment by KineticTheory** [comment]
Travis claims `The command "eval curl -sf  -o $CASHER_DIR/bin/casher https://build.travis-ci.org/files/casher " failed. Retrying, 2 of 3.` 

I will close and reopen to see if travis will run.

**Comment by alalazo** [comment]
Restarted the failing build.

**Pull request merged**